storagemarket/impl/dtutils/dtutils.go

```diff
@@ -82,6 +87,16 @@ func ClientDataTransferSubscriber(deals EventReceiver) datatransfer.Subscriber {
 
 		// data transfer events for progress do not affect deal state
 		switch event.Code {
+		case datatransfer.Restart:
+			err := deals.Send(voucher.Proposal, storagemarket.ClientEventDataTransferRestarted)
+			if err != nil {
+				log.Errorf("processing dt event: %w", err)
+			}
+		case datatransfer.Accept:
+			err := deals.Send(voucher.Proposal, storagemarket.ClientEventDataTransferInitiated, channelState.ChannelID())
```

Comment:
@hannahhoward  

The reason I have moved this here is because if we restart a failed transfer on the client side, we need to be sure that the provider has already created the channel which is what `Accept` will tell us.

Let me know what you think.